### PR TITLE
Hive JDBC Storage Handler: Incorrect results fetched from BOOLEAN and TIMESTAMP DataType From JDBC Data Source

### PR DESCRIPTION
**Pull request by shameersss1**

### CODE REVIEW

File: common/src/java/org/apache/hadoop/hive/common/type/Timestamp.java [hunk 1]
```diff
@@ -88,6 +88,10 @@ public Timestamp(Timestamp t) {
     this(t.localDateTime);
   }
 
+  public Timestamp(java.sql.Timestamp ts) {
```
Comment (jcamachor) [inline common/src/java/org/apache/hadoop/hive/common/type/Timestamp.java]: Could you remove this new constructor and use existing methods in `Timestamp` class to create the object (maybe you can encapsulate logic in utility method in JdbcSerDe)? `java.sql.Timestamp` is used in some places in Hive with UTC time zone vs system time zone and this constructor may make things confusing.

File: jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java [hunk 2]
```diff
@@ -226,7 +226,7 @@ public Object deserialize(Writable blob) throws SerDeException {
         case TIMESTAMP:
           if (rowVal instanceof java.sql.Timestamp) {
             java.sql.Timestamp timestampRowVal = (java.sql.Timestamp) rowVal;
-            rowVal = Timestamp.ofEpochMilli(timestampRowVal.getTime(), timestampRowVal.getNanos());
+            rowVal = new Timestamp(timestampRowVal);
```
Comment (jcamachor) [inline jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java]: It seems `Date` type should have same issue (visible depending on time zone). Could we change that accordingly too?

File: ql/src/test/results/clientpositive/llap/external_jdbc_table_partition.q.out [hunk 3]
```diff
@@ -101,10 +101,10 @@ POSTHOOK: query: SELECT * FROM jdbc_partition_table1 where ikey > 1
 POSTHOOK: type: QUERY
 POSTHOOK: Input: default@jdbc_partition_table1
 #### A masked pattern was here ####
-3	4000	120.0	25.4	hello	2.718	2017-06-05	2011-11-11 02:00:08
```
Comment (jcamachor) [inline ql/src/test/results/clientpositive/llap/external_jdbc_table_partition.q.out]: Cool... Not sure how we did not catch this before.

File: common/src/java/org/apache/hadoop/hive/common/type/Timestamp.java (hunk 1 above)
Comment (shameersss1) [reply common/src/java/org/apache/hadoop/hive/common/type/Timestamp.java]: Make Sense. Done

File: jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java (hunk 2 above)
Comment (shameersss1) [reply jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java]: Done

File: jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java [hunk 4]
```diff
@@ -217,16 +221,16 @@ public Object deserialize(Writable blob) throws SerDeException {
...
           } else {
             rowVal = Date.valueOf (rowVal.toString());
           }
           break;
         case TIMESTAMP:
           if (rowVal instanceof java.sql.Timestamp) {
-            java.sql.Timestamp timestampRowVal = (java.sql.Timestamp) rowVal;
-            rowVal = new Timestamp(timestampRowVal);
+            LocalDateTime localDateTime = ((java.sql.Timestamp) rowVal).toLocalDateTime();
+            rowVal = Timestamp.ofEpochSecond(localDateTime.toEpochSecond(UTC));
```
[the remark this answers is not in the document]
Comment (shameersss1) [reply jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java]: oops! my bad i missed it. Added!

File: jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java [hunk 5]
```diff
@@ -217,16 +221,16 @@ public Object deserialize(Writable blob) throws SerDeException {
...
           } else {
             rowVal = Date.valueOf (rowVal.toString());
           }
           break;
         case TIMESTAMP:
           if (rowVal instanceof java.sql.Timestamp) {
-            java.sql.Timestamp timestampRowVal = (java.sql.Timestamp) rowVal;
-            rowVal = Timestamp.ofEpochMilli(timestampRowVal.getTime(), timestampRowVal.getNanos());
+            LocalDateTime localDateTime = ((java.sql.Timestamp) rowVal).toLocalDateTime();
+            rowVal = Timestamp.ofEpochSecond(localDateTime.toEpochSecond(UTC), localDateTime.getNano());
```
Comment (shameersss1) [inline jdbc-handler/src/main/java/org/apache/hive/storage/jdbc/JdbcSerDe.java]: @jcamachor ^^

**Pull request closed without merging**